### PR TITLE
Feature/tao 6176/import eligibilities

### PR DESCRIPTION
**Pull request by ionutpad**
- [ ] https://github.com/oat-sa/tao-core/pull/1719

### CODE REVIEW

File: scripts/update/Updater.php [hunk 1]
```diff
@@ -159,5 +166,80 @@ public function update($initialVersion)
...
+                EligibilityCsvImporterFactory::OPTION_MAPPERS => [
+                    'default' => [
+                        EligibilityCsvImporterFactory::OPTION_MAPPERS_IMPORTER => new RdsEligibilityImportService()
+                    ]
+                ]
+            ]);
+
+            $this->getServiceManager()->register(EligibilityCsvImporterFactory::SERVICE_ID, $service);
+
+            $this->setVersion('3.12.0');
```
Comment (siwane) [inline scripts/update/Updater.php:244]: Should output 3.11.0

**Review by siwane — CHANGES_REQUESTED** [review]

File: controller/TestCenterManager.php [hunk 2]
```diff
@@ -106,7 +110,32 @@ public function editCenter()
...
+        /** @var EligibilityCsvImporterFactory $service */
+        $service = $this->getServiceLocator()->get(EligibilityCsvImporterFactory::SERVICE_ID);
+        $propertyKey = $this->getImportMapperTestCenterProperty();
+        foreach ($files as $file){
+            $report = $service->getImporter('default')->import($file['tmp_name'],[
+                $propertyKey => $testCenter
+            ]);
+        }
+
+        $data['report'] = $report;
```
Comment (SergiiTao) [inline controller/TestCenterManager.php:138]: if `empy($files)` `$report` is undefined

File: controller/TestCenterManager.php [hunk 3]
```diff
@@ -106,7 +110,32 @@ public function editCenter()
...
+        if (!$request->isPost()) {
+            throw new \Exception('Only post method allowed');
+        }
+        $files = $_FILES;
+
+        $testCenter = $this->getCurrentInstance();
+        /** @var EligibilityCsvImporterFactory $service */
+        $service = $this->getServiceLocator()->get(EligibilityCsvImporterFactory::SERVICE_ID);
+        $propertyKey = $this->getImportMapperTestCenterProperty();
+        foreach ($files as $file){
```
Comment (SergiiTao) [inline controller/TestCenterManager.php]: are you sure that $files is an array?

File: controller/TestCenterManager.php [hunk 4]
```diff
@@ -287,4 +321,38 @@ protected function getRequestOptions() {
...
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     */
+    protected function getImportMapperTestCenterProperty()
+    {
+        /** @var EligibilityCsvImporterFactory $service */
+        $service = $this->getServiceLocator()->get(EligibilityCsvImporterFactory::SERVICE_ID);
+        $mapper  = $service->getImporter('default')->getMapper();
+
+        $schema = $mapper->getOption(ImportMapper::OPTION_SCHEMA);
+        if(isset($schema[ImportMapper::OPTION_SCHEMA_MANDATORY])){
```
Comment (SergiiTao) [inline controller/TestCenterManager.php]: if (!isset(...

File: controller/TestCenterManager.php [hunk 5]
```diff
@@ -287,4 +321,38 @@ protected function getRequestOptions() {
         );
 
     }
+
+    /**
+     * @throws \common_exception_NotFound
+     * @throws \oat\oatbox\service\exception\InvalidService
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
```
Comment (SergiiTao) [inline controller/TestCenterManager.php:328]: Please add @return tag

File: controller/TestCenterManager.php [hunk 6]
```diff
@@ -287,4 +321,38 @@ protected function getRequestOptions() {
...
+            if (is_array($propertyKey) && count($propertyKey) === 1){
+                $valueMapper = reset($propertyKey);
+                if ($valueMapper instanceof RdsValidatorValueMapper){
+                    $class = $valueMapper->getOption(RdsValidatorValueMapper::OPTION_CLASS);
+                }
+            } else {
+                $class = $propertyKey;
+            }
+
+            if (TestCenterService::CLASS_URI === $class){
```
Comment (SergiiTao) [inline controller/TestCenterManager.php:355]: and what if not? seems it is possible that function will return null in this case

File: model/import/RdsEligibilityImportService.php [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        /** @var core_kernel_classes_Resource $testCenter */
+        $testCenter = $properties[EligibilityService::PROPERTY_TESTCENTER_URI];
+        /** @var core_kernel_classes_Resource $delivery */
+        $delivery = $properties[EligibilityService::PROPERTY_DELIVERY_URI];
+
+        $eligibility = $this->getEligibilityService()->getRootClass()->searchInstances(array(
+            EligibilityService::PROPERTY_TESTCENTER_URI => $testCenter,
+            EligibilityService::PROPERTY_DELIVERY_URI => $delivery
+        ));
+        $created = false;
```
Comment (SergiiTao) [inline model/import/RdsEligibilityImportService.php]: Unused variable

File: model/import/RdsEligibilityImportService.php [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+        if (count($eligibility) === 0){
+            $created = $this->getEligibilityService()->createEligibility($testCenter, $delivery);
+            $eligibility = $this->getEligibilityService()->getRootClass()->searchInstances(array(
+                EligibilityService::PROPERTY_TESTCENTER_URI => $testCenter,
+                EligibilityService::PROPERTY_DELIVERY_URI => $delivery
+            ));
+        }
+        $eligibility = current($eligibility);
```
Comment (SergiiTao) [inline model/import/RdsEligibilityImportService.php]: One row ago `$eligibility` was an array and now an item of this array.
Please don't use one name for diffrerent objects. It is difficult to debug

Comment (SergiiTao) [inline model/import/RdsEligibilityImportService.php]: what is current element? [0]  or this array was iterated somewhere? Please use [0] if you want the first item

File: model/import/RdsEligibilityImportService.php [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+     * @return array
+     */
+    protected function formatData(array $data, array $extraProperties)
+    {
+        return $data;
+    }
+
+    /**
+     * @param ImportMapper $mapper
+     * @return mixed
```
Comment (SergiiTao) [inline model/import/RdsEligibilityImportService.php]: @return `core_kernel_classes_Resource` ?

**Review by SergiiTao — CHANGES_REQUESTED** [review]

**Review by SergiiTao — APPROVED** [review]

**Pull request merged**